### PR TITLE
Fixes Teleport Talisman Exploit

### LINKED ISSUE #7636: Cultists can use teleport talisman while stunned
**Problem Description**:
As a cultist, I can run around with a teleport talisman primed ready to teleport, even when stunned.

**What did you expect to happen**: 
I expected to not be able to teleport while tazed, while not holding the teleport talisman.

**What happened instead**:
I was able to cheese the fuck out of security by teleporting when tazed.

**Why is this bad/What are the consequences:**
This is an EXTREMELY abuse-able cheese that is very unhealthy to the game, and seems like an unintended interaction.

**Steps to reproduce the problem**:
Make a teleport rune, make teleport talisman, use it in hand, put it in pocket. Select your teleport destination whenever you feel like it, don't know if it works while restrained/asleep but it works while stunned.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Not sure if it works while sleeping/dead/cuffed, but it definitely works while stunned.

### PR DESCRIPTION
**Pull request by mrturkeytoe**
Imsxz told me and Birdtalon about this issue on the discord where you can use a teleport talisman even while stunned, cuffed (or dead) without it being in an active hand and I wanted to try and learn some DM so Birb walked me through this fix. 

Please don't kill me if it's horrible this is my first PR

Edit: fixes #7636 (forgot that :v)

:cl: Mrturkeytoe
fix: Cult tele talismans can no longer be used while cuffed/dead/not in an active hand.
:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
Make sure you add "fixes #7636" to your post in there.

I kind of think this kind of check would be better above the `input()` so you don't pick a rune before you find out you can't. I can't code well though

File: code/game/gamemodes/cult/talisman.dm [hunk 1]
```diff
@@ -147,6 +147,11 @@
 		return ..(user, 0)
 
 	var/input_rune_key = input(user, "Choose a rune to teleport to.", "Rune to Teleport to") as null|anything in potential_runes //we know what key they picked
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/talisman.dm]:
```
	var/input_rune_key = input(user, "Choose a rune to teleport to.", "Rune to Teleport to") as null|anything in potential_runes //we know what key they picked
	var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key] //what rune does that key correspond to?
	if(!src || qdeleted(src) || !user || user.l_hand != src && user.r_hand != src || user.incapacitated() || !actual_selected_rune)
		return ..(user, 0)
```

99% of the time you don't want to check for `stat || restrained()`

**Comment by Fox-McCloud** [comment]
Please actually give a useful/descriptive title to your PRs.

File: code/game/gamemodes/cult/talisman.dm (hunk 1 above)
Comment (Birdtalon) [reply code/game/gamemodes/cult/talisman.dm]: True but one of the issues here is that if you open the input box and then die you can still teleport. If a cultist was using an anti drop implant they could still do so while dead.

**Comment by Birdtalon** [comment]
@taukausanake no it wouldn't. The problem here is that opening the input halts the code and waits for a user input. So you can open the input box and ignore it in complete safety. Then go and fight and teleport at any time as the code will run when you select a rune regardless of your state.

File: code/game/gamemodes/cult/talisman.dm (hunk 1 above)
Comment (Fox-McCloud) [reply code/game/gamemodes/cult/talisman.dm]: `incapacitated` checks for that.

Comment (Birdtalon) [reply code/game/gamemodes/cult/talisman.dm]: 👍

**Comment by tigercat2000** [comment]
@mrturkeytoe Could you please update this as detailed in Fox's review?

**Comment by mrturkeytoe** [comment]
@tigercat2000 Remove || Restrained()? Sure.

**Comment by tigercat2000** [comment]
@mrturkeytoe I meant, rather, changing it to this:
```dm
	var/input_rune_key = input(user, "Choose a rune to teleport to.", "Rune to Teleport to") as null|anything in potential_runes //we know what key they picked
	var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key] //what rune does that key correspond to?
	if(!src || qdeleted(src) || !user || user.l_hand != src && user.r_hand != src || user.incapacitated() || !actual_selected_rune)
		return ..(user, 0)
```

**Comment by Fox-McCloud** [comment]
Please implement this properly, @mrturkeytoe

**Comment by mrturkeytoe** [comment]
I'm not exactly sure I follow how you want this in the file, do you want the line that was edited and the one below it merged with this change you made? @Fox-McCloud

**Comment by mrturkeytoe** [comment]
You want it to replace `	var/input_rune_key = input(user, "Choose a rune to teleport to.", "Rune to Teleport to") as null|anything in potential_runes //we know what key they picked
	if(!input_rune_key || (!user.is_in_active_hand(src) || user.stat))
		to_chat(user, "<span class='warning'>You must have the talisman in a active hand to operate it!</span>")
		log_game("Teleport talisman failed - not in active hand or user restrained or dead")
	return ..(user, 0)

	var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key] //what rune does that key correspond to?
	if(!actual_selected_rune)
		return ..(user, 0)`

With this, correct? `var/input_rune_key = input(user, "Choose a rune to teleport to.", "Rune to Teleport to") as null|anything in potential_runes //we know what key they picked
	var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key] //what rune does that key correspond to?
	if(!src || qdeleted(src) || !user || user.l_hand != src && user.r_hand != src || user.incapacitated() || !actual_selected_rune)
		to_chat(user, "<span class='warning'>You must have the talisman in a active hand to operate it!</span>")
		log_game("Teleport talisman failed - not in active hand or user restrained or dead")
		return ..(user, 0)`

**Comment by Fox-McCloud** [comment]
need to get rid of all that you added, and copy paste this:

```
	var/input_rune_key = input(user, "Choose a rune to teleport to.", "Rune to Teleport to") as null|anything in potential_runes //we know what key they picked
	var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key] //what rune does that key correspond to?
	if(!src || qdeleted(src) || !user || user.l_hand != src && user.r_hand != src || user.incapacitated() || !actual_selected_rune)
		return ..(user, 0)
```

over line 149

**Comment by mrturkeytoe** [comment]
That gave two compile warnings of duplicate "var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key]" in the code.

**Comment by Fox-McCloud** [comment]
You obviously wouldn't have the definition right after having the definition.

Lines 155, 156, and 157 shouldn't exist.

**Comment by Fox-McCloud** [comment]
have 2 days to fix and test this; then this will be closed.

**Pull request closed without merging**